libzkbob-rs/src/merkle.rs

```diff
@@ -694,23 +885,148 @@ impl<D: KeyValueDB, P: PoolParams> MerkleTree<D, P> {
+        let new_root = self.get_node_full_batched(constants::HEIGHT as u32, 0, &mut update_batch, new_next_index);
+        self.db.write(update_batch).unwrap();
+
+        if self.next_index() < self.get_last_stable_index() {
+            self.set_last_stable_index(None);
+        }
+
+        new_root
+    }
+
+    // The rollback helper: build node (e.g. root) within 
+    // nodes before right_index (at level 0)
+    // It's supposed that the current tree has no nodes
+    // which subtree placed to the right of right_index
+    // Returns None in case of lack of nodes
+    // NOTE: as get_virtual_node_full routine, but not virtual one :)
+    //       it operates only with the currently filled nodes and update
+    //       the transaction batch to update the database
+    fn get_node_full_batched(
+        &mut self,
+        height: u32,
+        index: u64,
+        batch: &mut DBTransaction,
+        right_index: u64,
+    ) -> Option<Hash<P::Fr>> {
+        let node_left = index * (1 << height);
+        //let node_right = (index + 1) * (1 << height);
+        if node_left >= right_index {
+            //print!("💬[{}.{}]  ", height, index);
+            return Some(self.default_hashes[height as usize]);
         }
 
-        result
+        match self.get_opt(height, index) {
+            Some(hash) => {
+                //print!("✅[{}.{}]  ", height, index);
+                Some(hash)
+            },
+            None => {
+                if height > 0 {
+                    let left_child = self.get_node_full_batched(
+                        height - 1,
+                        2 * index,
+                        batch,
+                        right_index,
+                    );
+                    let right_child = self.get_node_full_batched(
+                        height - 1,
+                        2 * index + 1,
+                        batch,
+                        right_index,
+                    );
+
+                    if let (Some(left_child), Some(right_child)) = 
+                        (left_child, right_child)
+                    {
+                        let pair = [left_child, right_child];
+                        let hash = poseidon(pair.as_ref(), self.params.compress());
+
+                        //print!("🧮[{}.{}]={}.{}+{}.{}  ", height, index, height - 1, index << 1, height - 1, (index << 1) + 1);
+                        
+                        self.set_batched(batch, height, index, hash, 0);
+
+                        Some(hash)
+                    } else {
+                        //print!("❌[{}.{}]  ", height, index);
+                        None
+                    }
+                } else {
+                    //print!("❌[{}.{}]  ", height, index);
+                    None
+                }
+            }
+        }
+    }
+
+    pub fn wipe(&mut self) {
+        let mut wipe_batch = self.db.transaction();   
+        
+        // ???: It works fine in the local tests,
```

Comment:
Yes, kvdb-web implementation of DeletePrefix doesn't look correct. 
![image](https://user-images.githubusercontent.com/17165678/207031232-2516e9f4-2546-4a1b-a501-de27a5c16d63.png)